api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -755,17 +748,69 @@ public Long getCountOfPatients(String query, boolean includeVoided) {
-		
-		// Using Hibernate projections did NOT work here, the resulting queries could not be executed due to
-		// missing group-by clauses. Hence the poor man's implementation of counting search results.
-		//
-		return (long) criteria.list().size();
+
+		return (long) findPatients(query, includeVoided).size();
 	}
-	
-	/**
+
+    private List<Patient> findPatients(String query, boolean includeVoided) {
+        query = LuceneQuery.escapeQuery(query);
+
+        LuceneQuery<PatientIdentifier> luceneQuery = getPatientIdentifierLuceneQuery(query, includeVoided);
+		List<PatientIdentifier> patientIdentifiers = luceneQuery.list();
+		LuceneQuery<PersonName> luceneQuery2 = getPersonNameLuceneQuery(query, includeVoided);
+		luceneQuery2.skipSame(luceneQuery.listProjection("patient.personId"), "person.personId");
+		List<PersonName> personNames = luceneQuery2.list();
+		LuceneQuery<PersonAttribute> luceneQuery3 = getPersonAttributeLuceneQuery(query, includeVoided);
+		luceneQuery3.skipSame(luceneQuery.listProjection("patient.personId"), "person.personId");
+		luceneQuery3.skipSame(luceneQuery2.listProjection("person.personId"), "person.personId");
+		List<PersonAttribute> personAttributes = luceneQuery3.list();
+
+		List<Patient> personList = new LinkedList<>();
+		patientIdentifiers.forEach(patientIdentifier -> personList.add(patientIdentifier.getPatient()));
+		personNames.forEach(personName -> personList.add(Context.getPatientService().getPatient(personName.getPerson().getId())));
+		personAttributes.forEach(personAttribute -> personList.add(Context.getPatientService().getPatient(personAttribute.getPerson().getId())));
+
+		return personList;
+    }
+
+    private LuceneQuery<PatientIdentifier> getPatientIdentifierLuceneQuery(String query, boolean includeVoided) {
+        StringBuilder newQuery = new StringBuilder();
+        newQuery.append("identifier:\"").append(query).append("\"");
+        LuceneQuery<PatientIdentifier> luceneQuery = LuceneQuery
+                .newQuery(PatientIdentifier.class, sessionFactory.getCurrentSession(), newQuery.toString());
+        if(!includeVoided){
+            luceneQuery.include("voided", false);
+        }
+        luceneQuery.skipSame("patient.personId");
+        return luceneQuery;
+    }
+
+    private LuceneQuery<PersonName> getPersonNameLuceneQuery(String query, boolean includeVoided) {
+        StringBuilder newQuery = new StringBuilder();
+        newQuery.append("givenName:\"").append(query).append("\" OR familyName:\"").append(query)
+                .append("\" OR middleName:\"").append(query).append("\"");
+        LuceneQuery<PersonName> luceneQuery = LuceneQuery
+                .newQuery(PersonName.class, sessionFactory.getCurrentSession(), newQuery.toString());
+        if(!includeVoided){
+            luceneQuery.include("voided", false);
+        }
+        luceneQuery.skipSame("person.personId");
```

Comment:
I did that because those are separate methods and if someone will want to use it in the future there will be no duplicates.